src/syscheckd/create_db.c

```diff
@@ -88,44 +64,33 @@ void fim_generate_delete_event(fdb_t *fim_sql,
     cJSON_Delete(json_event);
 }
 
-void fim_delete_file_event(fdb_t *fim_sql,
-                           fim_entry *entry,
-                           pthread_mutex_t *mutex,
-                           void *_evt_data,
-                           __attribute__((unused)) void *_unused_field_1,
-                           __attribute__((unused)) void *_unused_field_2) {
-    event_data_t *evt_data = (event_data_t *)_evt_data;
-    directory_t *configuration = NULL;
-
-    configuration = fim_configuration_directory(entry->file_entry.path);
+int fim_generate_delete_event(const char *file_path,
+                              const void *_evt_data,
+                              const void *configuration) {
+    const directory_t *original_configuration = (const directory_t *)configuration;
+    get_data_ctx ctx = {
+        .event = (event_data_t *)_evt_data,
+        .config = original_configuration,
+        .path = file_path
+    };
+    callback_context_t callback_data;
+    callback_data.callback = process_delete_event;
+    callback_data.context = &ctx;
 
-    if (configuration == NULL) {
-        return;
+    if (original_configuration->options & CHECK_SEECHANGES) {
+        fim_diff_process_delete_file(file_path);
     }
-    /* Don't send alert if received mode and mode in configuration aren't the same.
-       Scheduled mode events must always be processed to preserve the state of the agent's DB.
-    */
-    switch (evt_data->mode) {
-    case FIM_REALTIME:
-        if (!(configuration->options & REALTIME_ACTIVE)) {
-            return;
-        }
-        break;
 
-    case FIM_WHODATA:
-        if (!(configuration->options & WHODATA_ACTIVE)) {
-            return;
-        }
-        break;
+    // Remove path from the DB.
+    int retVal = fim_db_remove_path(file_path);
 
-    default:
-        break;
+    if (retVal != FIMDB_ERR) {
+        retVal = fim_db_get_path(file_path, callback_data);
```

Comment:
I think you need to change the order of those functions calls regarding the extraction and deletion of the file.

Reply:
Yes, this is a bug ! I am fixing that ! Thank you !